src/main/java/crazypants/enderio/teleport/TravelController.java

```diff
@@ -611,25 +619,99 @@ public boolean isTravelItemActive(EntityPlayer ep) {
         return getTravelItemTravelSource(ep) != null;
     }
 
-    /** Returns null if no travel item is equipped. */
+    /** Returns null if no travel item is in inventory/baubles. */
     @Nullable
     public TravelSource getTravelItemTravelSource(EntityPlayer ep) {
-        if (ep == null || ep.getCurrentEquippedItem() == null) {
+        if (ep == null) {
             return null;
         }
+
         ItemStack equipped = ep.getCurrentEquippedItem();
-        if (equipped.getItem() instanceof ItemTeleportStaff) {
-            if (((ItemTeleportStaff) equipped.getItem()).isActive(ep, equipped)) {
-                return TravelSource.TELEPORT_STAFF;
-            }
-        } else if (equipped.getItem() instanceof IItemOfTravel) {
-            if (((IItemOfTravel) equipped.getItem()).isActive(ep, equipped)) {
-                return TravelSource.STAFF;
+        if (equipped == null || equipped.getItem() == null || !(equipped.getItem() instanceof IItemOfTravel)) {
+            equipped = findTravelItemInInventoryOrBaubles(ep);
+        }
+
+        if (equipped != null) {
+            if (equipped.getItem() instanceof ItemTeleportStaff) {
+                if (((ItemTeleportStaff) equipped.getItem()).isActive(ep, equipped)) {
+                    return TravelSource.TELEPORT_STAFF;
+                }
+            } else if (equipped.getItem() instanceof IItemOfTravel) {
+                if (((IItemOfTravel) equipped.getItem()).isActive(ep, equipped)) {
+                    return TravelSource.STAFF;
+                }
             }
         }
+
         return null;
     }
 
+    /**
+     * Returns null if no Travel item found in inventory/baubles. <br>
+     * DO NOT CHANGE THIS CODE WITHOUT ALSO CHANGING
+     * TravelController.{@link #findTravelItemSlotInInventoryOrBaubles(EntityPlayer)}
+     */
+    @Nullable
+    public ItemStack findTravelItemInInventoryOrBaubles(EntityPlayer ep) {
+        ItemStack travelItem = null;
+        for (int i = 0; i < ep.inventory.getSizeInventory(); i++) {
+            ItemStack stack = ep.inventory.getStackInSlot(i);
+            if (stack != null && stack.getItem() != null && (stack.getItem() instanceof IItemOfTravel)) {
+                travelItem = stack;
+                break;
+            }
+        }
+
+        if (travelItem == null) {
```

Comment:
As far as I can see you attempt to use Baubles as *optional* dependency. However methods in this class use Baubles API directly (that'll cause crash if Baubles mod isn't loaded).
I suggest to split this method into 2: for regular inventory and for Baubles one. And call second method only if Baubles mod is present.

Another solution would be to make Baubles mod a *hard* dependency (undesirable).

Reply:
If you read the BaublesUtil.instance().getBaubles(ep) code, it returns null if Baubles is not loaded. Furthermore, no crash happens in actual testing when loading up and using staff in all ways without Baubles installed.
![proofnobaubles](https://github.com/GTNewHorizons/EnderIO/assets/16872377/bd7b5042-4855-4752-b603-298f78baf010)
![image](https://github.com/GTNewHorizons/EnderIO/assets/16872377/759d9b25-05a1-4f4f-8f17-d39280a05a78)